### DISCUSSION TITLE
Jest config question

### DISCUSSION THREAD

**Discussion by daddyschmack** [nrwl/nx#10852] [General]
Hi All,

If this is the wrong place to post this, please advise.

I've inherited an NX project that is currently running Jasmine, and I was trying to turn on Jest, I'm an NX rookie, but I am aware I need to set Jest as the runner in the angular.json.. But before making Jest the default test runner, I wanted to actually run the tests.. so when I type in "Jest", I get the below error.

To add to the confusion, 
there is only one application listed in the angular.json that matches tha name:
```
"shared-material": {
      "projectType": "library",
      "root": "libs/shared/material",
```

However, running "nx run shared-material:test"  does not error out.. so maybe it's a configuration issue..  but I can't find the where that config is referenced more than once in the angular.json.. (there is no projects.json)

```
Error: Whoops! Two projects resolved to the same config path: /Users/johnjaeckle/Sites/UI-2/libs/shared/material/jest.config.js:

  Project 1: /Users/<username>/Sites/<app-name>/libs/shared/material
  Project 2: /Users/j<username>/Sites/<app-name>/libs/shared/material

This usually means that your "projects" config includes a directory that doesn't have any configuration recognizable by Jest. Please fix it.

    at ensureNoDuplicateConfigs (/Users/<username>/Sites/<app-name>/node_modules/@jest/core/node_modules/jest-config/build/index.js:395:13)
    at readConfigs (/Users/xxxxx/Sites/xxx/node_modules/@jest/core/node_modules/jest-config/build/index.js:472:5)
    at async runCLI (/Users/xxx/Sites/xxx/node_modules/@jest/core/build/cli/index.js:132:59)
    at async Object.run (/Users/jxxxx/Sites/xxx/node_modules/jest/node_modules/jest-cli/build/cli/index.js:155:37)

```

here is an excerpt from the current angular.json.. Is there a plugin or utility to convert this to jest?  Or am i going to have to figure 

    "shared-maps": {
      "projectType": "library",
      "root": "libs/shared/maps",
      "sourceRoot": "libs/shared/maps/src",
      "prefix": "app-name-maps",
      "architect": {
        "test": {
          "builder": "@angular-devkit/build-angular:karma",
          "options": {
            "main": "libs/shared/maps/src/test.ts",
            "tsConfig": "libs/shared/maps/tsconfig.spec.json",
            "karmaConfig": "libs/shared/maps/karma.conf.js"
          }
        },
        "lint": {
         .....
        }
      }
    },
    "shared-material": {
      "projectType": "library",
      "root": "libs/shared/material",
      "sourceRoot": "libs/shared/material/src",
      "prefix": "localact",
      "architect": {
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "libs/shared/material/jest.config.js",
            "passWithNoTests": true
          }
      

this out by trial and error? 
(If it were a single project, I would know how to convert)

Thanks,

John

**Comment by uvwild** [comment]
just ran into the same issue ;)
i have similar projects where i works and cannot find the differing config responsible for this either.
dep versions, jest.config. etc are all the same.
let me know please if this is solved

**Comment by rfprod** [comment]
> so when I type in "Jest", I get the below error.

@daddyschmack when you type in "Jest" where exactly? Do you mean when you execute the `jest` command in the terminal from the project root?

If so, have you tried the following from the project root?

```bash
npx nx run-many --target test --all
```

**Comment by blcirre** [comment]
Hi all, 
I am running into the same issue but most strange for me is that both paths indicated in the error are exactly the same. How could be referencing to having 2 projects resolved to same config path and both are the same project?
Thanks.
